lib/gds_api/locations_api.rb

```diff
@@ -0,0 +1,28 @@
+require_relative "base"
+require_relative "exceptions"
+
+class GdsApi::LocationsApi < GdsApi::Base
+  # Get a list of local custodian codes for a postcode
+  #
+  # @param [String, nil] postcode The postcode for which the custodian codes are requested
+  #
+  # @return [Array] All local custodian codes for a specific postcode
+  def local_custodian_code_for_postcode(postcode)
+    response = get_json("#{endpoint}/locations?postcode=#{postcode}.json")
+
+    return [] if response.nil?
```

Comment:
We should have a test for this condition (and also for `response.nil?` on line 26 of this file). There's already quite a lot going on in the first commit, so I'd add these `if`/`unless` guards, and their tests, in a second commit.